app/models/role.rb

```diff
@@ -232,7 +232,11 @@ def self.override_search_operator(operator)
   private
 
   def sync_inheriting_filters
-    self.filters.where(:override => false).each { |f| f.inherit_taxonomies! }
+    filters_check = []
+    self.filters.where(:override => false).each { |f| filters_check.push(f.save) }
+    errors.add :base, N_('Some of associated filters is invalid which prevented the role to be saved') if filters_check.include? false
+    errors.empty?
+    raise ActiveRecord::Rollback, N_("Unable to submit role: Problem with associated filter") if filters_check.include? false
```

Comment:
How about showing the error by finding the filter that failed then printing `#{filter.errors}`?